java/arcs/core/data/CapabilitiesNew.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+/** Store capabilities containing a grouping of individual capabilities. */
+class CapabilitiesNew(val ranges: List<CapabilityNew.Range> = emptyList()) {
```

Comment:
Since `Range` also inhertis from `CapabilityNew`, why can't you have `List<CapabilityNew>`?

Also, why pass in a list? Can't you make these explicit members of the class?

```
class CapabilitiesNew(
  val persistence: Persistence, // or Range<Persistence> if Range is generic.
  val ttl: TTL,
  ...
)

Reply:
made the parameter `List<CapabilityNew>`, the data member is still ranges.
I don't want to list all capabilities explicitly because I don't want to have to update this class when a new capabilities is added and mostly it is a subset of capabilities, so any individual one is optional.

Reply:
What if you make the fields nullable?

```
class CapabilitiesNew(
  val persistence: Persistence? = null, // or Range<Persistence> if Range is generic.
  val ttl: TTL? = null,
  ...
)
```

You can construct the `CapabilityNew` with named parameters as `CapabilityNew(ttl = .., persistence =... )`. Also, due to the default value, the old uses will work fine too. 

Don't feel strongly about this, but I think the code might get simpler.

Reply:
I prefer not to, because in the current way i can generically iterate on the list when i do comparison etc, and it is better extendable.